CMakeLists.txt

```diff
@@ -177,9 +177,13 @@ endif()
 
 # uClibc *intentionally* crashes in duplocale(), at least as of:
 # https://github.com/ffainelli/uClibc/blob/266bdc1/libc/misc/locale/locale.c#L1322
-# So, if it looks like we're compiling for a system like that just disable 
+# So, if it looks like we're compiling for a system like that just disable
 # locale handling entirely.
-exec_program(${CMAKE_C_COMPILER} ARGS -dumpmachine OUTPUT_VARIABLE CMAKE_GNU_C_MACHINE)
+if (CMAKE_VERSION GREATER_EQUAL "3.28")
+  execute_process (COMMAND ${CMAKE_C_COMPILER} -dumpmachine OUTPUT_QUIET ERROR_QUIET OUTPUT_VARIABLE CMAKE_GNU_C_MACHINE)
```

Comment:
Setting OUTPUT_QUIET looks wrong.  We _want_ the output saved in the variable but the cmake docs say setting that turns output off.

Reply:
Sorry now fixed to `ECHO_OUTPUT_VARIABLE`.